src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -609,6 +612,14 @@ ucs_status_t uct_ib_mlx5_srq_init(uct_ib_mlx5_srq_t *srq, struct ibv_srq *verbs_
     srq->db  = srq_info.dv.dbrec;
     uct_ib_mlx5_srq_buff_init(srq, srq_info.dv.head, srq_info.dv.tail,
                               sg_byte_count, sge_num);
+
+    /* Initialize SRQ number */
+    dv.srq.in       = verbs_srq;
+    dv.srq.out      = &dvsrq;
+    dvsrq.comp_mask = MLX5DV_SRQ_MASK_SRQN;
+    mlx5dv_init_obj(&dv, MLX5DV_OBJ_SRQ);
+    srq->srq_num    = dvsrq.srqn;
```

Comment:
seems it fails to compile when there is no MLX5DV_SRQ_MASK_SRQN defined, on other hand we need this only when using DevX so i guess we can:
1. add MLX5DV_SRQ_MASK_SRQN to ib.m4 CHECK_DECLS
2. put this code under `#if`, if not - set srqn to 0
3. when using this value as input for devx command (uct_ib_mlx5_devx_create_qp/create_dct) assert that it's not 0

Otherwise - 👍

Reply:
done